encode.go

```diff
@@ -212,11 +212,68 @@ func (c *locationCache) getLocation(offset int) *time.Location {
 	return location
 }
 
+var infinityTsEnabled = false
+var infinityTsNegative time.Time
+var infinityTsPositive time.Time
+
+const (
+	_infinityString                 = "-infinity"
+	infinityString                  = "infinity"
+	infinityTsEnabledAlready        = "pg: infinity timestamp enalbed already"
+	infinityTsNegativeMustBeSmaller = "pg: infinity timestamp: negative value must be smaller (before) than positive"
+)
+
+/**
+ * If EnableInfinityTs is not called, "-infinity" and "infinity" will return
+ * []byte("-infinity") and []byte("infinity") respectively, and potentially
+ * cause error "sql: Scan error on column index 0: unsupported driver -> Scan pair: []uint8 -> *time.Time",
+ * unless the input param is `*interface{}` instead of `*time.Time`
+ *
+ * If EnableInfinityTs is called with negative >= positive, it will panic
```

Comment:
Ah.  That's a good point.

I was worried about someone who can't process infinities at all and just wants to display a nice error if it gets those back from the database, but I guess in such a situation you can already do that with interface{} and dealing with the time.Time/[]byte difference.